package.json

```diff
@@ -10,52 +10,49 @@
   ],
   "repository": {
     "type": "git",
-    "url": "https://github.com/tableau-mkt/github-data-connector.git"
-  },
-  "engines": {
-    "node": "4.8.4"
+    "url": "https://github.com/marketplacer/github-data-connector.git"
   },
   "scripts": {
     "start": "node app.js",
     "postinstall": "node node_modules/gulp/bin/gulp build"
   },
   "dependencies": {
-    "cookie-parser": "^1.4.3",
-    "debug": "^2.6.3",
-    "es6-promise-pool": "^2.4.4",
-    "express": "^4.15.2",
-    "jquery": "^3.1.1",
-    "js-base64": "^2.1.9",
-    "lodash": "^4.17.4",
-    "node-env-file": "^0.1.8",
-    "request-promise": "^4.1.1",
-    "uglifyify": "^3.0.4"
+    "axios": "*",
```

Comment:
You've set almost ever dependency as unlocked. This is pretty risky from a stability point of view, as any package could ship a major (breaking) change through a mere `npm install`, which would be very confusing for any maintainer.

Reply:
🤔 package-lock.json should take care of it. 
Upstream project doesn't have a lock file and package.json doesn't put a hard lock on version with simple `npm install <package>` that is why I have included lock file to make it the source of truth for installations. 

As per npm release 5.x onwards lock file is supposed to be committed in VCS to keep same subsequent installs.

[lock file precedence order](https://docs.npmjs.com/cli/v7/commands/npm-install#synopsis)

Reply:
> package.json doesn't put a hard lock on version with simple npm install <package>

Correct, but it did would have at least kept things within the major versions (that's what the `^` is doing in statements like `"browserify": "^14.1.0",`).

The lockfile is just a lockfile. It's not a specification on what versions we allow **upgrades** to.

To provide a simple example: let's imagine for a moment that the `jquery` dependency depends on `cookie-parser`. We also have `cookie-parser` specified as an explicit dependency in `package.json`. 

Using `*` as our version definitions, this would mean that if `jquery` had an update one day, which we merged in, it could, if it wanted, update the version of the shared `cookie-parser` dependency, potentially to a version which breaks things for us.

But finally: consistency is key. There's significant value in doing things the way the rest of our projects do things. None of our projects use wildcards for dependencies in `package.json`. I'm not sure I see a good reason for this one to be an exception.

Reply:
IMHO that is the whole concept of having a lock file (complete version lock even to the level of sub-dependencies) which makes any version pinning in package.json redundant. 😄 once we have as working lock file major or minor infact all version levels are defined in lock file ([unless we change package.json](https://marketplacer.slack.com/archives/C5P5N1BUM/p1631665487177000?thread_ts=1631662030.173900&cid=C5P5N1BUM)) 

Regarding question why I put wildcards in package.json
Well putting wildcard in package.json was a way to quickly resolve into working dependencies and also getting a "working" lock file for subsequent installs. (when npm install was run for the first time after git pull after specially with node 14 and removing obsolete dependencies)

I agree we can mature this project as we develop further in future mostly till now changes are done from investigative purpose as well as per [ticket 1092](https://github.com/marketplacer/operations/issues/1092)

Reply:
`axios` is obsolete and full of bugs. Please don't use it. Use `fetch` or an alternative library based on it.

Reply:
> You've set almost every dependency as unlocked. This is pretty risky from a stability point of view, as any package could ship a major (breaking) change through a mere `npm install`, which would be very confusing for any maintainer.

Ellis is right.

Reply:
@ozmoroz Actually repo was using request-promise which is based on request  and [request](https://github.com/request/request) is officially obsolete as marked on their github page.

So using axios ([considering atleast actively maintained](https://github.com/axios/axios)) was merely getting to the point of successful installation and kicking off development server so that further investigations and scope of required improvements can be determined.

Reply:
> 🤔 package-lock.json should take care of it.
> Upstream project doesn't have a lock file and package.json doesn't put a hard lock on version with simple `npm install <package>` that is why I have included lock file to make it the source of truth for subsequent installations.
> 
> As per npm release 5.x onwards lock file is supposed to be committed in VCS to keep same subsequent installs.
> 
> [lock file precedence order](https://docs.npmjs.com/cli/v7/commands/npm-install#synopsis)

Hi, Farhan.
Having versions in both `package.json` and the lock file is a *safety valve*. A lock file can get corrupted. it actually happens quite often during merges gone wrong, etc. If you have versions in `package.json`, then Yarn will do its best heuristics to repair it. If you don't have versions in `package.json`, then all is lost. And besides, you don't really get any benefits from not putting the versions into `package.json`, so why do it?

Reply:
> @ozmoroz Actually repo was using request-promise which is based on request and [request](https://github.com/request/request) is officially obsolete as marked on their github page.
> 
> So using axios ([considering atleast actively maintained](https://github.com/axios/axios)) was merely getting to the point of successful installation and kicking off development server so that further investigations and scope of required improvements can be determined.

Axios was a library of choice about 5 years ago. Then it got nearly abandoned and accumulated lots of security vulnerabilities. Although the development appears to pick up now, the trust is already broken.

Besides, we now have a modern native API with `fetch`, so there is really no reason to use Axios any more.